testbed/src/main/java/org/spdx/testbed/util/TestCaseFinder.java

```diff
@@ -0,0 +1,102 @@
+package org.spdx.testbed.util;
+
+import org.spdx.testbed.TestCase;
+import org.spdx.testbed.TestCaseCategory;
+import org.spdx.testbed.util.testClassification.TestName;
+import org.springframework.beans.factory.annotation.AnnotatedBeanDefinition;
+import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider;
+import org.springframework.core.type.filter.AnnotationTypeFilter;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * Utility class to dynamically find test cases. Works by scanning the classpath for classes
+ * annotated with special marker annotations defined in the testClassification package. Can scan
+ * for specific attribute values on the annotations as well.
+ */
+public class TestCaseFinder {
+
+    public List<TestCase> findTestCasesByCategories(List<TestCaseCategory> categories) {
+        var foundCases = new ArrayList<TestCase>();
+        for (var category : categories) {
+            foundCases.addAll(determineTestCases(category.getAnnotationClass()));
```

Comment:
Hm... just randomly noticed that this seems wrong. When specifying multiple categories, it should be the intersection (e.g.: specifying formats "json, yaml" and spdx version "2.3" should find all tests for 2.3 that concern either of the two formats - not all tests that concern either version 2.3 _or_ one of the two formats)

Reply:
Ah, when reading last week I thought that intersections just haven't been implemented yet^^

Reply:
Technically true - and as long as there is only one category (apart from `TestName` which is special anyway), it doesn't matter. But the current implementation is a union, which doesn't make sense

Reply:
Fixed